src/components/inputs/CPriceInput.vue

```diff
@@ -0,0 +1,158 @@
+<template>
+  <v-text-field
+    v-model="fieldValue"
+    :append-inner-icon="mdiCurrencyEur"
+    :label="label"
+    :variant="variant"
+    :disabled="disabled"
+    :rules="rules"
+    placeholder="0,00"
+    color="primary"
+    @input="onInputChange"
+  ></v-text-field>
+</template>
+
+<script setup lang="ts">
+import { mdiCurrencyEur } from "@mdi/js";
+import { ref, watch } from "vue";
+
+/**
+ * The currently entered price.
+ */
+const model = defineModel<number | undefined>({ required: true });
+
+/**
+ * The model for the input field.
+ *
+ * This field is just a ref and will only be updated from the field itself and the watcher of the `model`.
+ *
+ * A computed ref is not used because the parse functions will automatically remove any disallowed chars on change.
+ * This causes the `model`s value to not change and therefore, the computed not to be updated, which leads to the problem
+ * that disallowed chars are not removed anymore.
+ */
+const fieldValue = ref<string | undefined>();
+
+/**
+ * Watches the model to set the newest value to `fieldValue`.
+ *
+ * This one will only use `formatPrice` to also support database formats like `12` to be parsed to `12,00` instead of `0,12`.
+ */
+watch(
+  model,
+  () => {
+    // Update field value to fit format
+    fieldValue.value = model.value
+      ? model.value.toFixed(2).toString().replace(".", ",")
+      : undefined;
+  },
+  { immediate: true },
+);
+
+/**
+ * This component builds an input field for a price with simplified typing experience.
+ *
+ * The format of the price will always be `x,xx`.
+ */
+withDefaults(
+  defineProps<{
+    /**
+     * The label of this field.
+     */
+    label: string;
+
+    /**
+     * The variant of this field.
+     */
+    variant?:
+      | "underlined"
+      | "outlined"
+      | "filled"
+      | "solo"
+      | "solo-inverted"
+      | "solo-filled"
+      | "plain";
+
+    /**
+     * Whether the input fields should be disabled.
+     */
+    disabled?: boolean;
+
+    /**
+     * The rules to use to validate the password.
+     */
+    rules: ((password: string) => true | string)[];
+  }>(),
+  { variant: "outlined" },
+);
+
+/**
+ * Parses the given string value into a price with the format `x,xx`.
+ * The formatted price can easily be parsed into a float value with `stringToFloat`.
```

Comment:
`stringToFloat` is not used here

Reply:
Yea, it doesn't say so, its just an information

Reply:
Please add an empty line above that though

Reply:
todo

Reply:
deprecated